### ISSUE TITLE
Docker daemon stops accepting connections when vpnkit is under load

### ISSUE THREAD

**Issue by imavroukakis** [docker/for-mac#2841]
Please, check https://docs.docker.com/docker-for-mac/troubleshoot/.
Issues without logs and details cannot be debugged, and will be closed.

- [x] I have tried with the latest version of my channel (Stable or Edge)
- [x] I have included the Diagnostics ID in this Issue
- [x] I have submitted Diagnostics

### Expected behavior
Docker CLI remains responsive under load
### Actual behavior
Cannot perform any Docker CLI related commands
### Information

```
Docker for Mac: version: 18.04.0-ce-mac62 (7e2bb492d35dc35ec0037d8024b7e7be57fc72a9)
macOS: version 10.12.6 (build: 16G1314)
logs: /tmp/7B0E50A9-8EE3-48E7-946D-550D9BAFA7E9/20180424-130751.tar.gz
failure: docker ps failed: (Failure "docker ps: timeout after 10.00s")
[OK]     vpnkit
[OK]     virtualization hypervisor
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         docker ps failed
[OK]     disk

Diagnostics id: 7B0E50A9-8EE3-48E7-946D-550D9BAFA7E9
```
### Steps to reproduce the behaviour
Launch two containers , as part of a maven build (using fabric8 plugin). Containers are latest MSSQL and ElasticSearch 5.5.2 and start integration tests. At a specific point in the test suite, several hundred threads are running against the two services in the container, which eventually makes it unresponsive, with the following errors

```
➜  docker ps
Error response from daemon: Bad response from Docker engine

➜ docker stats
Error response from daemon: Bad response from Docker engine
```

When the test is stopped, Docker becomes responsive again

Both containers are allocated 3Gb of RAM each, Docker itself is set to use 6Gb

**Comment by imavroukakis** [comment]
This seems to happen when vpnkit reaches around 2001 open files
```
lsof -p $(pgrep vpnkit) | wc -l
    2001
```

**Comment by jpswade** [comment]
I also have the same issue:

```
$ kubectl cluster-info
Kubernetes master is running at https://localhost:6443

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
Unable to connect to the server: EOF

$ kubectl cluster-info dump
Unable to connect to the server: EOF
```

It's not trivial to debug and diagnose this.

**Comment by imavroukakis** [comment]
This also happens on the stable version

**Comment by imavroukakis** [comment]
Thanks @djs55 . Any way we can tweak that setting?

**Comment by djs55** [comment]
I don't think it's easily tweakable -- I think we would need to switch over from `select` to `kqueue` (or `poll`) in `hyperkit`'s hypervisor socket implementation to remove the limit. I've filed an issue in [moby/hyperkit] and will gather feedback / second opinions from other maintainers there.

**Comment by djs55** [comment]
@imavroukakis I have an experimental build which should bypass this particular scalability bottleneck. The build is available from 
https://download-stage.docker.com/mac/master/Docker.dmg

Let me know if it behaves better or worse. If worse, could you upload a fresh set of diagnostics and quote the diagnostics ID? Thanks!

**Comment by IhorHryshkov** [comment]
@djs55 I have equal problem and I trying your new version of docker: 18.09.0-beta5 and don't have this problem, waiting this fix in stable 18.09.0.

**Comment by imavroukakis** [comment]
Apologies for the late response, will check the build @djs55
